Source/Core/Core/HW/AddressSpace.cpp

```diff
@@ -68,7 +68,7 @@ Accessors::iterator Accessors::end() const
   return nullptr;
 }
 
-std::optional<u32> Accessors::Search(u32 haystack_start, u8* needle_start, u32 needle_size,
+std::optional<u32> Accessors::Search(u32 haystack_start, const u8* needle_start, u32 needle_size,
```

Comment:
You may want to also make `needle_size` be a `size_t`, rather than a `u32`, as it wouldn't require casting from external usages that need to interact with standard containers, etc. Ditto for the other member functions that accept a size.